### ISSUE TITLE
Datatype in `Setω` breaks unification on the LHS

### ISSUE THREAD

**Issue by L-TChen** [agda/agda#5730]
Consider the following example,

```agda
{-# OPTIONS --with-K #-}

open import Agda.Primitive        using (Setω)
open import Agda.Builtin.Equality using (_≡_; refl)

-- change `Set` to `Setω` breaks `seq`
data RecD  (I : Set) : Set where
  ι : (i : I) → RecD I

data RecO {I J : Set} (e : I → J) : RecD I → RecD J → Setω where
  ι : (i : I) (j : J) (eq : e i ≡ j) → RecO e (ι i) (ι j)

seq : {I J K : Set} {e : I → J} {f : J → K} {D : RecD I} {E : RecD J} {F : RecD K}
  → RecO f E F → RecO e D E
  → RecO (λ i → f (e i)) D F
seq (ι _ _ refl) (ι _ _ refl) = ι _ _ refl
```

where `RecD I` is just (equivalent to) `I` and `RecO e D E` consists of all pairs of `(i : I)` and `(j : J)` such that `e i ≡ j`. We can write a composition function `seq` to compose two pairs. The only thing unusual is that `RecO` lives in `Setω`.

Now, if the `RecD` is declared to be in `Setω` (so that `RecD` can be level-polymorphic), then Agda `seq` fails to solve the unification problem:

```
I'm not sure if there should be a case for the constructor ι,
because I get stuck when trying to solve the following unification
problems (inferred index ≟ expected index):
  ι i₁ ≟ D
  ι j ≟ ι i
when checking that the pattern ι i₁ .i refl has type RecO e D (ι i)
```

I don't really see why changing from `Set` to `Setω` breaks this unification problem even using `--with-K`. Any idea?

**Comment by jespercockx** [comment]
It seems the unifier is not up-to-date with datatypes in sorts other than `Set l`, i.e. `Setω` and `SSet`. Fix incoming.

**Comment by L-TChen** [comment]
Right, after inspecting the output of `tc.lhs.unify`, I see that while solving `ι i =?= D, ι j =?= ι j₁` the unifier jumps to the deletion step instead of the injectivity step.

**Comment by L-TChen** [comment]
Hum.. not so fast. By turning on `--without-K`, Agda now panics instead:
```
Panic: Pattern match failure in do expression at
src/full/Agda/TypeChecking/Primitive/Cubical.hs:2265:7-21
when checking the definition of seq
``` 
pointing to the function clause of `seq`, no matter `RecD` is in `Set` or `Setω`.

**Comment by L-TChen** [comment]
Ping @jespercockx. Do you have any idea about the above error after your fix? ~[ I can try to fix this next week. ]~

**Comment by jespercockx** [comment]
That's an error in the implementation of `--cubical`, I honestly have to idea what's going on but I guess it has something to do with the fact that it does not take into account datatypes in `Setω`. Perhaps @Saizan has a better idea.

**Comment by Saizan** [comment]
Yeah, the left-inverse machinery should more gracefully give up in case it would need to transport in `Setω`.

More long-term we should have `transp/hcomp` primitives that apply to the `Setω` hierarchy, since they are considered fibrant.

**Comment by jespercockx** [comment]
I changed the implementation so it throws a proper error message instead of a match failure. I'm not planning to do anything more on this.

**Comment by L-TChen** [comment]
This issue has been resolved by @plt-amy in https://github.com/agda/agda/commit/2b4be2cc1fff03adf1dfac04cea83f7ce940a86f.

**Comment by andreasabel** [comment]
This must have been fixed earlier even, the OP works fine in 2.6.2 (but not 2.6.1).  (Works even `--without-K`.)

**Comment by L-TChen** [comment]
> This must have been fixed earlier even, the OP works fine in 2.6.2 (but not 2.6.1). (Works even `--without-K`.)

Does the OP works with `--without-K` in 2.6.2?
